extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/conf/RadiusAuthenticationProtocol.java

```diff
@@ -19,100 +19,54 @@
 
 package org.apache.guacamole.auth.radius.conf;
 
+import org.apache.guacamole.properties.EnumGuacamoleProperty.PropertyValue;
+
 /**
  * This enum represents supported RADIUS authentication protocols for
  * the guacamole-auth-radius extension.
  */
 public enum RadiusAuthenticationProtocol {
     
     /**
-     * Password Authentication Protocol (PAP)
+     * Password Authentication Protocol (PAP).
      */
-    PAP("pap"),
+    @PropertyValue("pap")
+    PAP,
     
     /**
-     * Challenge-Handshake Authentication Protocol (CHAP)
+     * Challenge-Handshake Authentication Protocol (CHAP).
      */
-    CHAP("chap"),
+    @PropertyValue("chap")
+    CHAP,
     
     /**
-     * Microsoft implementation of CHAP, Version 1 (MS-CHAPv1)
+     * Microsoft implementation of CHAP, Version 1 (MS-CHAPv1).
      */
-    MSCHAPv1("mschapv1"),
+    @PropertyValue("mschapv1")
+    MSCHAP_V1,
     
     /**
-     * Microsoft implementation of CHAP, Version 2 (MS-CHAPv2)
+     * Microsoft implementation of CHAP, Version 2 (MS-CHAPv2).
      */
-    MSCHAPv2("mschapv2"),
+    @PropertyValue("mschapv2")
+    MSCHAP_V2,
     
     /**
-     * Extensible Authentication Protocol (EAP) with MD5 Hashing (EAP-MD5)
+     * Extensible Authentication Protocol (EAP) with MD5 Hashing (EAP-MD5).
      */
-    EAP_MD5("eap-md5"),
+    @PropertyValue("eap-md5")
+    EAP_MD5,
 
     /**
      * Extensible Authentication Protocol (EAP) with TLS encryption (EAP-TLS).
      */
-    EAP_TLS("eap-tls"),
+    @PropertyValue("eap-tls")
+    EAP_TLS,
 
     /**
      * Extensible Authentication Protocol (EAP) with Tunneled TLS (EAP-TTLS).
      */
-    EAP_TTLS("eap-ttls");
+    @PropertyValue("eap-ttls")
+    EAP_TTLS;
 
-    /**
-     * This variable stores the string value of the protocol, and is also
-     * used within the extension to pass to JRadius for configuring the
-     * library to talk to the RADIUS server.
-     */
-    private final String strValue;
-    
-    /**
-     * Create a new RadiusAuthenticationProtocol object having the
-     * given string value.
-     * 
-     * @param strValue
-     *     The value of the protocol to store as a string, which will be used
-     *     in specifying the protocol within the guacamole.properties file, and
-     *     will also be used by the JRadius library for its configuration.
```

Comment:
Eep - will deleting this break RADIUS? I don't recall the string value being used for JRadius, but if that's the case, I should definitely put some of this back.

Reply:
Okay, so, as long as it properly converts to a `String` using `toString()`, it should be okay:
```
        RadiusAuthenticator radAuth = radiusClient.getAuthProtocol(
                confService.getRadiusAuthProtocol().toString());
        
        if (radAuth == null)
            throw new GuacamoleException("Could not get a valid RadiusAuthenticator for specified protocol: " + confService.getRadiusAuthProtocol());
```
(https://github.com/apache/guacamole-client/blob/4e4b6f24f2d34a91f5381da955680f85c2ce56c3/extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java#L129-L133)

The Inner Protocol may have to be re-worked, a bit:

```
        // If we're using EAP-TTLS, we need to define tunneled protocol
        if (radAuth instanceof EAPTTLSAuthenticator) {
            RadiusAuthenticationProtocol innerProtocol =
                    confService.getRadiusEAPTTLSInnerProtocol();
            
            if (innerProtocol == null)
                throw new GuacamoleException("Missing or invalid inner protocol for EAP-TTLS.");


            ((EAPTTLSAuthenticator)radAuth).setInnerProtocol(innerProtocol.toString());
```
(https://github.com/apache/guacamole-client/blob/4e4b6f24f2d34a91f5381da955680f85c2ce56c3/extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java#L161-L170)

Reply:
Ah, but "properly" there means into whatever value is required by JRadius.

I'll restore a means of getting the authenticator name specific to JRadius from the protocol.

Reply:
Okay - I basically implemented it such that the value set in `guacamole.properties` is the same value that `JRadius` uses, so it just need to be able to retrieve that value and pass it through.  If there's a way to do that mapping back to the `@PropertyValue` annotation I'm not opposed to that.

Reply:
OK - I've added:

* A constructor parameter for the JRadius protocol name (referencing the specific `NAME` constant) and static property, `JRADIUS_PROTOCOL_NAME`, that exposes this.
* A convenience function, `getAuthenticator()`, which returns the `RadiusAuthenticator` instance for the protocol in question.

and carried that refactor through.